### PR TITLE
[482] Add tools for creating Ports with direction

### PR DESCRIPTION
**Pull request by jerome-obeo**
Bug: https://github.com/eclipse-syson/syson/issues/482

### CODE REVIEW

File: backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/ChildPartUsageNodeDescriptionProvider.java [hunk 1]
```diff
@@ -220,6 +223,49 @@ private NodeTool createNodeTool(NodeDescription nodeDescription, EClass eClass,
                 .build();
     }
 
+    private NodeTool createNodeToolWithDriection(NodeDescription nodeDescription, EClass eClass, NodeContainmentKind nodeKind, FeatureDirectionKind direction) {
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/ChildPartUsageNodeDescriptionProvider.java:226]: createNodeToolWithDriection => createNodeToolWithDirection

File: backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/FirstLevelChildPartUsageNodeDescriptionProvider.java [hunk 2]
```diff
@@ -224,6 +228,49 @@ private NodeTool createNodeTool(NodeDescription nodeDescription, EClass eClass,
...
+                .expression("aql:newFeatureMembership")
+                .children(createEClassInstance.build(), createView.build());
+
+        var createMembership = this.viewBuilderHelper.newCreateInstance()
+                .typeName(SysMLMetamodelHelper.buildQualifiedName(SysmlPackage.eINSTANCE.getFeatureMembership()))
+                .referenceName(SysmlPackage.eINSTANCE.getElement_OwnedRelationship().getName())
+                .variableName("newFeatureMembership")
+                .children(changeContexMembership.build());
+
+        var directionLabel = direction.getLiteral().substring(0, 1).toUpperCase() + direction.getLiteral().substring(1);
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/FirstLevelChildPartUsageNodeDescriptionProvider.java:265]: You're allowed to use `org.apache.commons.lang3.StringUtils#capitalize`

File: backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/RootDefinitionNodeDescriptionProvider.java [hunk 3]
```diff
@@ -182,6 +186,53 @@ private NodeTool createNodeTool(NodeDescription nodeDescription, EClass eClass,
...
+                .expression("aql:newFeatureMembership")
+                .children(createEClassInstance.build(), createView.build());
+
+        var createMembership = this.viewBuilderHelper.newCreateInstance()
+                .typeName(SysMLMetamodelHelper.buildQualifiedName(SysmlPackage.eINSTANCE.getFeatureMembership()))
+                .referenceName(SysmlPackage.eINSTANCE.getElement_OwnedRelationship().getName())
+                .variableName("newFeatureMembership")
+                .children(changeContexMembership.build());
+
+        var directionLabel = direction.getLiteral().substring(0, 1).toUpperCase() + direction.getLiteral().substring(1);
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/RootDefinitionNodeDescriptionProvider.java:227]: You're allowed to use `org.apache.commons.lang3.StringUtils#capitalize`

File: backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/RootUsageNodeDescriptionProvider.java [hunk 4]
```diff
@@ -182,6 +186,53 @@ private NodeTool createNodeTool(NodeDescription nodeDescription, EClass eClass,
...
+                .expression("aql:newFeatureMembership")
+                .children(createEClassInstance.build(), createView.build());
+
+        var createMembership = this.viewBuilderHelper.newCreateInstance()
+                .typeName(SysMLMetamodelHelper.buildQualifiedName(SysmlPackage.eINSTANCE.getFeatureMembership()))
+                .referenceName(SysmlPackage.eINSTANCE.getElement_OwnedRelationship().getName())
+                .variableName("newFeatureMembership")
+                .children(changeContexMembership.build());
+
+        var directionLabel = direction.getLiteral().substring(0, 1).toUpperCase() + direction.getLiteral().substring(1);
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/RootUsageNodeDescriptionProvider.java:227]: You're allowed to use `org.apache.commons.lang3.StringUtils#capitalize`

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/description/ToolDescriptionService.java [hunk 5]
```diff
@@ -190,4 +192,40 @@ public NodeTool createNodeToolFromDiagramBackground(NodeDescription nodeDescript
                 .build();
     }
 
+    public NodeTool createNodeToolFromDiagramWithDirection(NodeDescription nodeDescription, EClass eClass, FeatureDirectionKind direction, IDescriptionNameGenerator descriptionNameGenerator) {
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/description/ToolDescriptionService.java:195]: Is this method the same than the others in this PR?
If yes, could you share the code in a service class please?

File: backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/ChildPartUsageNodeDescriptionProvider.java (hunk 1 above)
Comment (AxelRICHARD) [inline backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/ChildPartUsageNodeDescriptionProvider.java:226]: Is this method the same than the others in this PR?
If yes, could you share the code in a service class please?

File: backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/FirstLevelChildPartUsageNodeDescriptionProvider.java [hunk 6]
```diff
@@ -224,6 +228,49 @@ private NodeTool createNodeTool(NodeDescription nodeDescription, EClass eClass,
                 .build();
     }
 
+    private NodeTool createNodeToolWithDirection(NodeDescription nodeDescription, EClass eClass, NodeContainmentKind nodeKind, FeatureDirectionKind direction) {
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/FirstLevelChildPartUsageNodeDescriptionProvider.java:231]: Is this method the same than the others in this PR?
If yes, could you share the code in a service class please?

File: backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/RootDefinitionNodeDescriptionProvider.java [hunk 7]
```diff
@@ -182,6 +186,53 @@ private NodeTool createNodeTool(NodeDescription nodeDescription, EClass eClass,
                 .build();
     }
 
+    private NodeTool createNodeToolWithDirection(NodeDescription nodeDescription, EClass eClass, NodeContainmentKind nodeKind, FeatureDirectionKind direction) {
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/RootDefinitionNodeDescriptionProvider.java:189]: Is this method the same than the others in this PR?
If yes, could you share the code in a service class please?

File: backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/RootUsageNodeDescriptionProvider.java [hunk 8]
```diff
@@ -182,6 +186,53 @@ private NodeTool createNodeTool(NodeDescription nodeDescription, EClass eClass,
                 .build();
     }
 
+    private NodeTool createNodeToolWithDirection(NodeDescription nodeDescription, EClass eClass, NodeContainmentKind nodeKind, FeatureDirectionKind direction) {
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/RootUsageNodeDescriptionProvider.java:189]: Is this method the same than the others in this PR?
If yes, could you share the code in a service class please?

File: backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/ChildPartUsageNodeDescriptionProvider.java (hunk 1 above)
Comment (jerome-obeo) [reply backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/ChildPartUsageNodeDescriptionProvider.java]: Done

File: backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/FirstLevelChildPartUsageNodeDescriptionProvider.java (hunk 2 above)
Comment (jerome-obeo) [reply backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/FirstLevelChildPartUsageNodeDescriptionProvider.java]: Done

File: backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/RootDefinitionNodeDescriptionProvider.java (hunk 3 above)
Comment (jerome-obeo) [reply backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/RootDefinitionNodeDescriptionProvider.java]: Done

File: backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/RootUsageNodeDescriptionProvider.java (hunk 4 above)
Comment (jerome-obeo) [reply backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/RootUsageNodeDescriptionProvider.java]: Done

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/description/ToolDescriptionService.java (hunk 5 above)
Comment (jerome-obeo) [reply backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/description/ToolDescriptionService.java:195]: Done

File: backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/FirstLevelChildPartUsageNodeDescriptionProvider.java (hunk 6 above)
Comment (jerome-obeo) [reply backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/FirstLevelChildPartUsageNodeDescriptionProvider.java]: Done

File: backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/RootDefinitionNodeDescriptionProvider.java (hunk 7 above)
Comment (jerome-obeo) [reply backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/RootDefinitionNodeDescriptionProvider.java]: Done

File: backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/RootUsageNodeDescriptionProvider.java (hunk 8 above)
Comment (jerome-obeo) [reply backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/RootUsageNodeDescriptionProvider.java]: Done

**Review by AxelRICHARD — APPROVED** [review]

**Pull request merged**